### PR TITLE
WIP. adding order, 1 new client router

### PR DESCRIPTION
**Pull request by alexanderyar**

### CODE REVIEW

File: src/controllers/auth/userEmailVerification.ts [hunk 1]
```diff
@@ -5,12 +5,25 @@ import { NotFound } from "http-errors";
...
+  const users = await User.find({
+    where: {
+      verification_token: verificationToken,
+    },
+    // FIXME relations doesn't work, so client ID can't be sent to frontEnd in redirect link
+    // relations: ["client"],
+  });
+  const user = users[0];
+  // console.log(users);
+
```
Comment (KossGuk) [inline src/controllers/auth/userEmailVerification.ts:26]: Lets roll back to annotation as it was

File: src/controllers/auth/userLogin.ts [hunk 2]
```diff
@@ -7,12 +7,44 @@ import { Unauthorized } from "http-errors";
...
 
+const { SECRET_KEY } = process.env;
+// const users = await userRepository.find({
+//   where: {
+//     /* conditions */
+//   },
+//   relations: {
+//     /* relations */
+//   },
+// });
```
Comment (KossGuk) [inline src/controllers/auth/userLogin.ts:21]: If we don't need this code in future lets keep the code clear

File: src/controllers/auth/userLogin.ts [hunk 3]
```diff
@@ -7,12 +7,44 @@ import { Unauthorized } from "http-errors";
...
 export const userLogin: RequestHandler = async (req, res) => {
   const { email, password } = req.body;
   const user = await User.findOneBy({ email });
+
+  // const users = await User.find({
+  //   where: {
```
Comment (KossGuk) [inline src/controllers/auth/userLogin.ts:27]: Here as well

File: src/controllers/auth/userRegistration.ts [hunk 4]
```diff
@@ -30,6 +32,27 @@ export const userRegistration: RequestHandler = async (req, res) => {
 
   await result.save();
```
Comment (KossGuk) [inline src/controllers/auth/userRegistration.ts:33]: Lets make this save in transaction as well.
(Save User and Client)

File: src/controllers/clients/clientsAddInfo.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import { RequestHandler } from "express";
+import Client from "../../entity/Client.entity";
+
+export const clientsAddInfo: RequestHandler = async (req, res) => {
+  const info = req.body;
+
+  console.log(info);
+
+  const client_id = req.user.client_id!;
```
Comment (KossGuk) [inline src/controllers/clients/clientsAddInfo.ts:9]: "client_id!"  Do we need '!' at the end? If yes for what?

**Review by KossGuk — CHANGES_REQUESTED** [review]

File: src/controllers/clients/clientsAddInfo.ts (hunk 5 above)
Comment (alexanderyar) [reply src/controllers/clients/clientsAddInfo.ts:9]: if this controller took over it means we do have client_id for sure. But in line 11 we need to update client info using client id. An alternative could be adding ! to client_id in .update func

Comment (KossGuk) [reply src/controllers/clients/clientsAddInfo.ts]: Oh. I didn't know about this functionality.

**Review by KossGuk — APPROVED** [review]

**Pull request merged**